Tofe/halium subtree

Tofee opened the pull request:
Use a subtree to include Halium's manifest into our manifest.

A git subtree can be updated separately, though the subtree history will appear in our history too.

jbruechert committed:
Add Halium/halium-devices

Change-Id: Ia03bd17b5f514e2d08689b1cef55a2beb33858b6

Commit:
Use halium fork of hardware/qcom/media-caf/msm8960

This is necessary, because the original repo used java files.

Herrie82 committed:
Add missing packages

These were used in Halium 5.1 but not in 7.1 but might still be needed iproute2 for example. Others added just in case.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

Herrie82 committed:
Add commonly used repos to default.xml (#15)

Some of them like android_device_qcom_sepolicy and android_device_qcom_common are used commonly in the majority of projects, so make sense to sync them as default. Others are small, so don't cause much overhead either.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

Commit:
Update android 7.1.1 r25 to android 7.1.2 r36 (#10)

Herrie82 committed:
default.xml: Add android_vendor_qcom_opensource_dataservices

Solves: CANNOT LINK EXECUTABLE "/system/bin/netmgrd": library
"librmnetctl.so" not found

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

bhush9 committed:
Add libcore

it seems new versions of frameworks_native have libgui which depends on
the libnativeloader and hence libcore.

It have java stuff removed

Change-Id: I63c6ba86c2341e4e4cb08f2d36d20a952a99dcdf

bhush9 committed:
Fix the libcore in manifest

Change-Id: Ice770b482d292f4ab6f8a6304bdd3f6974440fd9

UniversalSuperBox committed:
Add halium-boot to manifest (#23)

Herrie82 committed:
default.xml: Set sync-c value and increase sync-j value (#28)

sync-c will make sure only indicated branch in manifest will be downloaded. Will save on download space.

Tofee committed:
Add Redmi 5 (rosy) repositories

Change-Id: I8e6c53b05fbc234a8e031e57e7c2252260f5c61a
Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Tofee committed:
Include Halium's manifest

Change-Id: I805ce06d8fb3b0d85d00162e7e839ebcd6eb46dd
Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Tofee committed:
Add 'halium/' from commit 'cdd9a27847b3f7034a1782c124204b5675953824'

git-subtree-dir: halium
git-subtree-mainline: 384c79e61d1d8b72943b83caebf38fadec4bfe0e
git-subtree-split: cdd9a27847b3f7034a1782c124204b5675953824

Herrie82 commented:
@Tofee Here it seems we're duplicating Halium's default.xml. Or at least it shows that way in our subtree. Also we wouldn't need our targets here anymore. We could use Halium's. I sent all our targets (- Rosy) to Halium already.

Tofee commented:
It's not a duplication, but a subtree: the whole Halium history is merged with ours (which isn't really nice), but it's kept separate by git and we can remerge the subtree whenever we want to update the Halium stuff.
Submodules would look a bit nicer, but repo doesn't support it yet...

Herrie82 commented:
@Tofee This would still mean that we'd need to manually update the Halium bits right? I.e. keep an eye on upstream and merge when we feel like. This can be good (in case @bhush9 breaks something :P) or extra work, depending on how you look at it.

Tofee commented:
Yes, that's what that would mean, exactly. subtree is very alike a rebase,
but with upstream in a separate subfolder.
So... we try ?

I tried to think of a way to have delegates in halium-devices/setup to call
a script of ours, but I didn't find a good way to do this.

On Fri, Jun 29, 2018 at 10:46 AM Herrie <notifications@github.com> wrote:

> @Tofee <https://github.com/Tofee> This would still mean that we'd need to
> manually update the Halium bits right? I.e. keep an eye on upstream and
> merge when we feel like. This can be good (in case @bhush9
> <https://github.com/bhush9> breaks something :P) or extra work, depending
> on how you look at it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/webOS-ports/android/pull/4#issuecomment-401290838>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACYXiL38vNYMCgmYejbafJXt7K63v3-aks5uBelRgaJpZM4U8F-7>
> .
>

Herrie82 commented:
@Tofee not sure what you mean by "I tried to think of a way to have delegates in halium-devices/setup to call
a script of ours, but I didn't find a good way to do this." ?

Tofee commented:
Currently to initialize the local manifest for a device, we call
halium-devices/setup <device>.
But wouldn't it be nice if we could have a hook, somehow, and that this
script calls a script of ours which will, in turn, add a luneos_targets.xml
local manifest?
However I can't see a way to implement this in a generic way: a second,
optoinal, argument to call halium-devices/setup ? Or something based on the
environment ? I don't know.

On Fri, Jun 29, 2018 at 10:51 AM Herrie <notifications@github.com> wrote:

> @Tofee <https://github.com/Tofee> not sure what you mean by "I tried to
> think of a way to have delegates in halium-devices/setup to call
> a script of ours, but I didn't find a good way to do this." ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/webOS-ports/android/pull/4#issuecomment-401292118>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACYXiK4z1d5XenqqQ3AyzFVBnAemqZtHks5uBep7gaJpZM4U8F-7>
> .
>

Herrie82 commented on default.xml:
```diff
@@ -1,252 +1,60 @@
+  <include name="halium/default.xml"/>
+  
+  <remove-project name="Halium/hybris-boot" />
+
```
You would want to add something to drop halium-boot as well just in case

Herrie82 commented on default.xml:
```diff
@@ -1,252 +1,60 @@
+  <remove-project name="mer-hybris/android_system_core" />
+  <project path="system/core" name="webOS-ports/android_system_core" groups="pdk" revision="wop-12.1"/>
+ -->
+  <!-- luneos/luneos-hal: contain patches to apply before starting the build. -->
```
These can be dropped in general since halium-devices/setup script will take care of all this with regexps. If there are exceptions we should extend the setup script in halium-devices to cover that as well.

Herrie82 commented on default.xml:
```diff
@@ -1,252 +1,60 @@
+  <project path="system/core" name="webOS-ports/android_system_core" groups="pdk" revision="wop-12.1"/>
+ -->
+  <!-- luneos/luneos-hal: contain patches to apply before starting the build. -->
+  <project path="luneos/luneos-hal" name="android_luneos_hal" remote="webos-ports" revision="halium-7.1"/>
```
All below code is in halium-devices now, except for Rosy. Could be dropped

Tofee committed:
Cleanup manifest

All the other targets are now handled with
halium/devices/setup <device>.

However keep luneos_hal for now, as it is still needed today.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
